b2/bucket.py

```diff
@@ -393,6 +399,37 @@ def _upload_large_file(
         response = self.api.session.finish_large_file(file_id, part_sha1_array)
         return FileVersionInfoFactory.from_api_response(response)
 
+    def _find_unfinshed_file(self, upload_source, file_name, file_info, part_ranges):
+        for file in self.list_unfinished_large_files():
+            if file.file_name == file_name and file.file_info == file_info:
+                files_match = True
+                finished_parts = []
```

Comment:
I think it would make sense for finished_parts to be a dict that maps from part number to the part object.  That will make the lookup quicker during upload.